buildagent-generation-update-vmss.yml

```diff
@@ -25,7 +29,7 @@ jobs:
 
   variables:
   - name: ManagedImageName
-    value: $[stageDependencies.buildagent_template_vm.generate_image.outputs['setVars.ManagedImageNameOutput'] ]
+    value: $[stageDependencies.${{ parameters.build_stage_job_name }}.outputs['setVars.ManagedImageNameOutput'] ]
```

Comment:
Can we add the `.generate_image` back to this line? It's a bit cleaner so we aren't passing that from multiple places.